Occasional Java UnknownHostException at pod startup

Pods managed by seata-server StatefulSet sometimes fail to start. The detail error stack is as follow:

```
java.lang.RuntimeException: java.net.UnknownHostException: seata-server-0.seata-server-cluster: Name or service not known
	at io.seata.common.util.NetUtil.convertIpIfNecessary(NetUtil.java:330)
	at io.seata.common.util.NetUtil.isValidIp(NetUtil.java:309)
	at io.seata.server.Server.start(Server.java:68)
	at io.seata.server.ServerRunner.run(ServerRunner.java:60)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.lambda$callRunners$2(SpringApplication.java:749)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:483)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:744)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1289)
	at io.seata.server.ServerApplication.main(ServerApplication.java:30)
Caused by: java.net.UnknownHostException: seata-server-0.seata-server-cluster: Name or service not known
	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1330)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1283)
	at java.net.InetAddress.getAllByName(InetAddress.java:1199)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at java.net.InetAddress.getByName(InetAddress.java:1077)
	at io.seata.common.util.NetUtil.convertIpIfNecessary(NetUtil.java:328)
	... 18 common frames omitted
```

The underlying reason is that seata-server would call `NetUtil.convertIpIfNecessary` method to translate host name of the headless service subdomain, e.g., "seata-server-0.seata-server-cluster". However, the coredns service may not add the service name registry yet, leading to the occasional `UnknownHostException`.

--- a/controllers/stateful_set.go
+++ b/controllers/stateful_set.go
@@ -63,6 +63,22 @@ func initStatefulSet(s *seatav1alpha1.SeataServer) *appsv1.StatefulSet {
 	return statefulSet
 }
 
+const PythonScript = `
+import time
+import socket
+
+print('Waiting $SEATA_IP to be resolved...')
+while True:
+	try:
+		ip = socket.gethostbyname('$SEATA_IP')
+		print('Resolve $SEATA_IP to', ip)
+	except:
+		print('Cannot resolve $SEATA_IP, wait 2 seconds', flush=True)
+		time.sleep(2)
+		continue
+	exit(0)
+`
+
 func updateStatefulSet(ctx context.Context, statefulSet *appsv1.StatefulSet, s *seatav1alpha1.SeataServer) {
 	logger := log.FromContext(ctx)
 	var envs []apiv1.EnvVar
@@ -85,10 +101,9 @@ func updateStatefulSet(ctx context.Context, statefulSet *appsv1.StatefulSet, s *
 		},
 		Args: []string{
 			"-c",
-			fmt.Sprintf(
-				"export SEATA_IP=$(HOST_NAME).%s; /bin/bash /seata-server-entrypoint.sh;",
-				s.Spec.ServiceName,
-			),
+			fmt.Sprintf("export SEATA_IP=$(HOST_NAME).%s;", s.Spec.ServiceName) +
+				fmt.Sprintf("python3 -c \"\n%s\n\";", PythonScript) +
+				"/bin/bash /seata-server-entrypoint.sh;",
 		},
 		Ports: []apiv1.ContainerPort{
 			{Name: "service-port", ContainerPort: s.Spec.Ports.ServicePort},

